repose-aggregator/tests/performance-tests/roles/gatling/files/simulations/services/uriredaction/UriRedactionServiceSimulation.scala

```diff
@@ -0,0 +1,87 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package services.uriredaction
+
+import com.typesafe.config._
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import io.gatling.http.request.builder.HttpRequestBuilder
+
+import scala.concurrent.duration._
+import scala.util.Random
+
+/**
+ * Simple uriredaction performance simulation.
+ */
+class UriRedactionServiceSimulation extends Simulation {
+  // properties to configure the Gatling test
+  val conf = ConfigFactory.load("application.conf")
+  val confRoot = "test"
+  val throughput = conf.getInt(s"$confRoot.throughput")
+  val duration = conf.getInt(s"$confRoot.duration")
+  val warmUpDuration = conf.getInt(s"$confRoot.warmup_duration")
+  val rampUpUsers = conf.getInt(s"$confRoot.ramp_up_users.new_per_sec")
+  val rampUpDuration = conf.getInt(s"$confRoot.ramp_up_users.duration_in_sec")
+  val percentile3ResponseTimeUpperBound = conf.getInt(s"$confRoot.expectations.percentile3_response_time_upper_bound")
+  val percentSuccessfulRequest = conf.getInt(s"$confRoot.expectations.percent_successful_requests")
+
+  // this value is provided through a Java property on the command line when Gatling is run
+  val baseUrl = conf.getString("test.base_url")
+
+  val httpConf = http.baseURL(s"http://$baseUrl")
+
+  // set up the warm up scenario
+  val warmup = scenario("Warmup")
+    .forever() {
+      exec(getRequest)
+    }
+    .inject(
+      constantUsersPerSec(rampUpUsers) during(rampUpDuration seconds))
+    .throttle(
+      jumpToRps(throughput), holdFor(warmUpDuration minutes),  // warm up period
+      jumpToRps(0), holdFor(duration minutes))                 // stop scenario during actual test
+
+  // set up the main scenario
+  val mainScenario = scenario("URI Redaction Service Test")
+    .forever() {
+      exec(getRequest)
+    }
+    .inject(
+      nothingFor(warmUpDuration minutes),  // do nothing during warm up period
+      constantUsersPerSec(rampUpUsers) during(rampUpDuration seconds))
+    .throttle(
+      jumpToRps(throughput), holdFor((warmUpDuration + duration) minutes))
+
+  // run the scenarios
+  setUp(
+    warmup,
+    mainScenario
+  ).assertions(
+    global.responseTime.percentile3.lte(percentile3ResponseTimeUpperBound),
+    global.successfulRequests.percent.gte(percentSuccessfulRequest)
+  ).protocols(httpConf)
+
+  def getRequest: HttpRequestBuilder = {
+    http(session => session.scenario)
+      .get("/v2/admin/redactMe/required")
+      .check(status.is(200))
```

Comment:
Do we also want a sanity check to prove that the redaction was processed as expected? I know functional tests should prove that, but here it would just be to make sure that the performance test is set up correctly.

Reply:
How would you check that? It would be in the communication with the `OpenTracing` service.

Reply:
Fair point. Not sure what I was thinking there...